tests/sdk/v2/test_workflow.py

```diff
@@ -259,3 +260,112 @@ def test_is_exportable_to_dot(tmp_path):
 
         # Then
         assert dot_path.exists()
+
+
+FULL_RESOURCES = {
+    "cpu": "2",
+    "memory": "32Gi",
+    "disk": "1Ti",
+    "gpu": "1",
+    "nodes": 10,
+}
+
+NONE_RESOURCES = {
+    "cpu": None,
+    "memory": None,
+    "gpu": None,
+    "disk": None,
+    "gpu": None,
+    "nodes": None,
+}
+
+
+class TestResources:
+    @pytest.fixture
+    def resourced_workflow(self):
+        def _fixture(**kwargs):
+            @sdk.workflow(resources=sdk.Resources(**kwargs))
```

Comment:
**thought**: I wonder why `mypy` didn't complain about types of dict values here. I would expect we'd need a `TypedDict` here as well.

Reply:
I'm not sure!